tests/persons_tests.py

```diff
@@ -1,24 +1,193 @@
 from unittest import TestCase
-from classes.persons import Persons
+from rooms.dojo import Dojo
+from persons.staffs import Staff
+from persons.fellows import Fellow
+from persons.persons import Person
+import sys
+
 
 class TestPersons(TestCase):
-    def test_add_person_successfully(self):
-        new_person = Persons()
-        new_staff = new_person.add_person("Andy Carroll", "staff")
-        self.assertIsInstance(Staff)
-        self.assertEqual(new_staff.name,"Andy Carroll")
-        self.assertIsInstance(Employee)
-        new_fellow = new_person.add_person("Jeremy Johnson", "fellow","Y")
-        self.assertIsInstance(Fellow)
-        self.assertEqual(new_fellow.name,"Jeremy Johnson")
-        self.assertIsInstance(Employee)
-        self.assertEqual(new_fellow.livingspace,"")
-
-    def test_add_person_invalid_input(self):
-        new_person = Persons()
-        new_staff = new_person.add_person("Andy Carroll", "staff", "Y")
-        self.assertEqual(new_staff,"Staff cannot request for a livingspace!")
-        new_staff = new_person.add_person("  ", "staff")
-        self.assertEqual(new_staff,"Staff cannot be created with empty name!")
-        new_staff = new_person.add_person("Samora Dake","type","Y")
-        self.assertEqual(new_staff,"Person cannot br created due to invalid designation!")
+    def setUp(self):
+        self.dojo = Dojo()
+
+    def tearDown(self):
+        self.dojo.reset()
+
+    def test_add_staff(self):
+        """ This function test for successful adding of staff """
+        new_staff = self.dojo.add_person("Andy Carroll", "staff")
+        self.assertIsInstance(new_staff, Staff)
+        self.assertEqual(new_staff.name, "Andy Carroll")
+        self.assertIsInstance(new_staff, Person)
+
+    def test_add_fellow(self):
+        """ This function test for successful adding of fellow """
+        new_fellow = self.dojo.add_person("Jeremy Johnson", "fellow", "Y")
+        self.assertIsInstance(new_fellow, Fellow)
+        self.assertEqual(new_fellow.name, "Jeremy Johnson")
+        self.assertIsInstance(new_fellow, Person)
+        self.assertEqual(new_fellow.livingspace, "")
+
+    def test_add_person_invalid_input_1(self):
+        """ This function checks for invalid inputs """
+        new_staff = self.dojo.add_person("Andy Carroll", "staff", "Y")
+        output = sys.stdout.getvalue().strip()
+        self.assertEqual(output, "Staff cannot request for a livingspace!")
+
+    def test_add_person_invalid_input_2(self):
+        new_staff = self.dojo.add_person("  ", "staff")
+        output = sys.stdout.getvalue().strip()
+        self.assertEqual(output, \
+                    "Person cannot be created with an empty name!")
+
+    def test_add_person_invalid_input_3(self):
+        new_staff = self.dojo.add_person("Samora Dake", "type", "Y")
+        output = sys.stdout.getvalue().strip()
+        self.assertEqual(output, \
+                    "Person cannot be created due to invalid designation!")
+
```

Comment:
For the above three tests that check for invalid inputs, please provide more descriptive function names. `test_add_person_invalid_input_1` is not cool.